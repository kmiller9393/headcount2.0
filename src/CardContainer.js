import React from 'react';
import Card from './Card';
import CompareCard from './CompareCard';
import PropTypes from 'prop-types';
import './App.css';
import './Card.css';

const CardContainer = ({
  filteredDistricts,
  chooseCard,
  compareDistricts,
  compareCards,
  comparedAverages,
  removeCard
}) => {
  const newCard = filteredDistricts.map((district, index) => {
    return (
      <Card
        location={district.location}
        stats={district.stats}
        key={index}
        chooseCard={chooseCard}
        compareCards={compareCards}
        removeCard={removeCard}
      />
    );
  });

  const selectedDistricts = compareDistricts.map((district, index) => {
    return (
      <Card
        location={district.location}
        stats={district.stats}
        key={index}
        chooseCard={chooseCard}
        compareCards={compareCards}
        compareDistricts={compareDistricts}
        selected
        removeCard={removeCard}
      />
    );
  });

  return (
    <React.Fragment>
      <div className="Selected">
        {selectedDistricts[0]}
        {compareDistricts.length > 1 && (
          <CompareCard
            DistrictOne={Object.keys(comparedAverages)[0]}
            DistrictOneData={Object.values(comparedAverages)[0]}
            comparison={Object.values(comparedAverages)[2]}
            DistrictTwo={Object.keys(comparedAverages)[1]}
            DistrictTwoData={Object.values(comparedAverages)[1]}
          />
        )}
        {selectedDistricts[1]}
      </div>
      <div className="App">{newCard}</div>
    </React.Fragment>
  );
};

CardContainer.propTypes = {
  filteredDistricts: PropTypes.array,
  chooseCard: PropTypes.func,
  compareDistricts: PropTypes.array,
  compareCards: PropTypes.func,
  comparedAverages: PropTypes.object,
  removeCard: PropTypes.func
};

export default CardContainer;
