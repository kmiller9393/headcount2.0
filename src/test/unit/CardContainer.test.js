import React from 'react';
import CardContainer from '../../CardContainer';
import { shallow, mount } from 'enzyme';

describe('CardContainer', () => {
  let wrapper;
  let removeCardMock;
  let chooseCardMock;
  let compareCardsMock;
  let filteredDistricts = [];
  let compareDistricts = [];
  let comparedAverages = {};
  beforeEach(() => {
    removeCardMock = jest.fn();
    chooseCardMock = jest.fn();
    compareCardsMock = jest.fn();
    wrapper = shallow(
      <CardContainer
        removeCard={removeCardMock}
        chooseCard={chooseCardMock}
        compareCards={compareCardsMock}
        filteredDistricts={filteredDistricts}
        compareDistricts={compareDistricts}
        comparedAverages={comparedAverages}
      />
    );
  });
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
