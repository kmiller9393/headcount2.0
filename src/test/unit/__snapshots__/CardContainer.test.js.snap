// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    chooseCard={[Function]}
    compareCards={[Function]}
    compareDistricts={Array []}
    comparedAverages={Object {}}
    filteredDistricts={Array []}
    removeCard={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="Selected"
>
          
</div>,
        <div
          className="App"
>
          
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            false,
            undefined,
          ],
          "className": "Selected",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          false,
          undefined,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "App",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="Selected"
>
            
</div>,
          <div
            className="App"
>
            
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              false,
              undefined,
            ],
            "className": "Selected",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            false,
            undefined,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "App",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
